### ISSUE TITLE
Option to decode JSON as object, instead of associative array

### ISSUE THREAD

**Issue by rodamaral** [halaxa/json-machine#15]
By default, `json_decode` will convert the string to an object, not to associative arrays. I wonder if it's possible to get the same behaviour in json-machine, or should I use `json_decode(json_encode($field), false)`?

**Comment by halaxa** [comment]
Hello, thanks for your question. This feature is not planned to be implemented. The reason is, that this can be easily done using native PHP type casting.

# Short version
```php
foreach(JsonMachine::fromFile('some.json') as $array) {
    $object = (object) $array; // array casts to stdClass instance, the same as json_decode produces.
    echo $object->example;
}
```
That's it.

# Longer version
If you want to have your consumer code clean without having to cast the array every time as in previous example, you can make yourself a little generator function:
```php
function toObjects(iterable $items): Generator
{
    foreach ($items as $item) {
        yield (object) $item;
    }
}
```
and use it like this:
```php
foreach (toObjects(JsonMachine::fromFile('some.json')) as $object) {
    echo $object->example;
}
```
Please let me know if it solves your problem.

**Comment by rodamaral** [comment]
It solves our problem. Thank you for your library.

**Comment by halaxa** [comment]
Glad to help. If you like it, give it a star, share it and spread the word :).

You made a valid point. If you want to be a contributor, you could make pull request with this example in README.md. What do you think?

**Comment by halaxa** [comment]
I made a function for that called `objects`. See cbc45bd

**Comment by halaxa** [comment]
Object will be default in version 1.0. See #37.

**Comment by halaxa** [comment]
`objects` function is deprecated since 0.4.0

**Comment by halaxa** [comment]
Done in `master` branch. Will be released in 1.0.0.
